### ISSUE TITLE
Return order of nonzero elements differs from NumPy

### ISSUE THREAD

**Issue by manopapad** [nv-legate/cunumeric#50]
Consider the following program:

```
a = np.arange(25).reshape((5,5))
print(np.nonzero(a > 17))
```

When ran with python NumPy (or legate.numpy on 1 CPU) the non-zero indices are returned in this order:

```
x = [3 3 4 4 4 4 4]
y = [3 4 0 1 2 3 4]
```

If instead we run with legate.numpy on 4 CPUs (using `NUMPY_TEST` to force legate.numpy to do distributed execution) (command line: `NUMPY_TEST=1 legate nz-order.py -lg:numpy:test --cpus 4`) we get:

```
x = [4 4 4 3 3 4 4]
y = [0 1 2 3 4 3 4]
```

I.e. legate.numpy returns the indices grouped by tile (see [nz-order.pdf](https://github.com/nv-legate/legate.numpy/files/6729441/nz-order.pdf) for a visualization), instead of returning them according to the global row-major order, as is guaranteed in the [NumPy API](https://numpy.org/doc/stable/reference/generated/numpy.nonzero.html). This is a side effect of how distributed nonzero is implemented. Making this work like NumPy would require a sort after every nonzero call.

We could simply decide to live with this incompatibility, since I expect most code using nonzero will not explicitly depend on the order that nonzero elements are returned in. The most likely scenario I can think of where this incompatibility would be problematic is if the user code mixes the results of different nonzero calls in the same operation:

```
import legate.numpy as np

# too small to be partitioned; indices will be in C order
small = np.ones((2,2))
small_is = np.nonzero(small)

# large enough to be partitioned; indices will be grouped by tile.
large = np.zeros((10000,10000))
large[2500,2500] = 2.0
large[2500,7500] = 3.0
large[2501,2500] = 4.0
large[2501,7500] = 5.0
large_is = np.nonzero(large)

small[small_is] = large[large_is]
print(small)
```

**Comment by manopapad** [comment]
Another way to handle this would be to repartition the source array in C order before collecting the nonzero elements within each partition. Then the order of nonzero indices in the overall output would naturally be  be in C order.

**Comment by magnatelee** [comment]
this issue has already been addressed by #216.
